packages/blocks/src/embed-block/image/image-block.ts

```diff
@@ -117,17 +117,52 @@ export class ImageBlockComponent extends NonShadowLitElement {
   @state()
   _source!: string;
 
+  @state()
+  _uploadingStatus!: string;
+
   // This is the initial width before event resize is applied
 
+  async setImageSource() {
+    const storage = await this.model.page.blobs;
+    assertExists(storage);
+    try {
+      const url = await storage.get(this.model.sourceId);
+      url && (this._source = url);
+    } catch (e) {
+      this._uploadingStatus = 'finish';
+      this._source = '';
+    }
+  }
+
   override async firstUpdated() {
     this.model.propsUpdated.on(() => this.requestUpdate());
     this.model.childrenUpdated.on(() => this.requestUpdate());
+    this._uploadingStatus = 'loading';
     // exclude padding and border width
     const { width, height } = this.model;
-    const storage = await this.model.page.blobs;
-    assertExists(storage);
-    const url = await storage.get(this.model.sourceId);
-    url && (this._source = url);
+    if (this.model.page.awarenessAdapter.isUploading(this.model.sourceId)) {
+      this._uploadingStatus = 'loading';
+      this.model.page.awarenessAdapter.signals.update.subscribe(
```

Comment:
call the dispose function on `disconnectedCallback `

Reply:
ok